packages/blocks/src/api/parser/apply-built-in-validation-fixes.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * Internal dependencies
+ */
+import { fixCustomClassname } from './fix-custom-classname';
+
+/**
+ * Attempts to fix block invalidation by applying build-in validation fixes
+ * like moving all extra classNames to the className attribute.
+ *
+ * @param {WPBlock}                               block     block object.
+ * @param {import('../registration').WPBlockType} blockType Block type. This is normalize not necessary and
+ *                                                          can be inferred from the block name,
+ *                                                          but it's here for performance reasons.
+ *
+ * @return {WPBlock} Fixed block object
+ */
+export function applyBuiltInValidationFixes( block, blockType ) {
```

Comment:
Right now this only "fixes" the className issues, potentially, we could consider enhancing this in the future by trying to map `style` attributes changes with the different customization options in Gutenberg and the `style` block attribute.